site/en/blog/chrome-116-beta-whats-new-for-extensions/index.md

```diff
@@ -0,0 +1,139 @@
+</figure>
+
+## Programmatically open a Sidepanel {: #sidepanel-open }
+
+Sidepanel has been one of the most requested features in Chrome extensions and has been available in Chrome since version 114. After launching the Side Panel API, one of the first pieces of feedback that we’ve received was that developers want a way to programmatically open a side panel. And here it is: [`chrome.sidePanel.open`](chrome-sidepanel-open) is now in beta. You can use it to open the extension side panel programmatically in response to a user interaction, such as a context menu click:
+
+```js
+chrome.contextMenus.onClicked.addListener((info, tab) => {
+  if (info.menuItemId === 'openSidePanel') {
+    // This will open the panel in all the pages on the current window.
+    chrome.sidePanel.open({ windowId: tab.windowId });
+  }
+});
+```
+
+## WebSocket support in Service Workers {: #sw-websockets }
+
+WebSocket support is critical for many extensions planning to move to Manifest V3. Chrome 116 further improves WebSocket support in service workers as all WebSocket activity will reset the [30s service worker idle timer][sw-lifecycle]. This means that as long as your WebSocket is active, the service worker will stay alive. 
+
+You can use this to implement a keepalive mechanism ensuring your service worker stays active while you’re waiting for messages from your server - even if it takes more than 30s until the next message arrives:
+
+```js
+function keepAlive() {
+  const keepAliveIntervalId = setInterval(
+    () => {
+      if (webSocket) {
+        webSocket.send('keepalive');
+      } else {
+        clearInterval(keepAliveIntervalId);
+      }
+    },
+    // It's important to pick an interval that's shorter than 30s, to
+    // avoid that the service worker becomes inactive.
+    20 * 1000
+  );
+}
+```
+
+Check out our new WebSocket [guide](websocket-guide) and [sample](websocket-sample) for more details.
+
+## Strong keepalive for Service Workers {: #sw-keepalive }
+
+Speaking of service worker lifecycle, another important update has landed: strong keepalive for APIs requiring user interaction. APIs that require a user interaction will have "strong" keepalives for extension service workers (i.e., allow the worker to take longer than 5 minutes on this task):
+
+* [`permissions.request()`][permission-request]
+* [`desktopCapture.chooseDesktopMedia()`][desktop-capture]
+* [`identity.launchWebAuthFlow()`][identity-launchwebflow]
+* [`management.uninstall()`][management-uninstall]
+
+## Recording audio and video in the background {: #tab-capture }
+
+Another gap between Manifest V2 and Manifest V3 has been closed: you can record audio and video in the background using `tabCapture` and offscreen documents. Use the [`chrome.tabCapture`][tabcapture] API in a service worker
+to obtain a stream ID following a user gesture. This can then be passed to an [offscreen document](offscreen-document) to start recording.
```

Comment:
```suggestion
To obtain a stream ID following a user gesture. This can then be passed to an [offscreen document](offscreen-document) to start recording.
```